rotkehlchen/chain/evm/decoding/thegraph/decoder.py

```diff
@@ -58,6 +81,73 @@ def _decode_delegator_staking(self, context: DecoderContext) -> DecodingOutput:
+            return self._decode_delegation_transferred_to_l2(context)
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _get_user_address(self, address: ChecksumEvmAddress) -> ChecksumEvmAddress | None:
+        """Get the user address (benficiary) from the vesting contract if it exists"""
+        if self.evm_inquirer.get_code(address) != '0x':
+            try:
+                raw_beneficiary_address = self.base.evm_inquirer.call_contract(
+                    contract_address=address,  # the vesting contract
+                    abi=GRAPH_TOKEN_LOCK_WALLET_BENEFICIARY_ABI,
+                    method_name='beneficiary',
+                )
+                return deserialize_evm_address(raw_beneficiary_address)
+            except RemoteError as e:
+                log.error(f'During delegation transfer to arbitrum got error calling {address} beneficiary(): {e!s}')  # noqa: E501
+                return None
+        return address
+
+    def _decode_delegation_transferred_to_l2(self, context: DecoderContext) -> DecodingOutput:
+        """Decode a transfer delegating GRT from ethereum to arbitrum"""
+        user_address = self._get_user_address(hex_or_bytes_to_address(context.tx_log.topics[1]))
+        if not user_address or not self.base.is_tracked(user_address):
+            return DEFAULT_DECODING_OUTPUT
+        delegator_l2 = hex_or_bytes_to_address(context.tx_log.topics[2])
+        indexer = hex_or_bytes_to_address(context.tx_log.topics[3])
+        l2_indexer = hex_or_bytes_to_address(context.tx_log.data[:32])
+        transferred_delegation_tokens = token_normalized_value(
+            token_amount=hex_or_bytes_to_int(context.tx_log.data[32:]),
+            token=self.token,
+        )
+        event = self.base.make_event_from_transaction(
+            transaction=context.transaction,
+            tx_log=context.tx_log,
+            event_type=HistoryEventType.INFORMATIONAL,
+            event_subtype=HistoryEventSubType.NONE,
+            asset=self.token,
+            balance=Balance(amount=FVal(transferred_delegation_tokens)),
+            notes=f'Delegation of {transferred_delegation_tokens} GRT transferred from indexer {indexer} to L2 indexer {l2_indexer}.',  # noqa: E501
+            location_label=context.transaction.from_address,
+            counterparty=CPT_THEGRAPH,
+            address=context.transaction.to_address,
+            extra_data={'delegator_l2': delegator_l2},
+        )
+        return DecodingOutput(event=event)
+
+    def _decode_contract_deposit(self, context: DecoderContext) -> DecodingOutput:
+        """Decode a deposit of ETH to cover the arbitrum fees of delegating GRT"""
+        user_address = self._get_user_address(hex_or_bytes_to_address(context.tx_log.topics[1]))
+        if not user_address or not self.base.is_tracked(user_address):
+            return DEFAULT_DECODING_OUTPUT
+        indexer = hex_or_bytes_to_address(context.tx_log.topics[1])
```

Comment:
newline after returns please

```suggestion

        indexer = hex_or_bytes_to_address(context.tx_log.topics[1])
```